implementations/rust/ockam/ockam_command/src/markdown/mod.rs

```diff
@@ -256,31 +256,38 @@ fn generate_arg_markdown(buffer: &mut Vec<u8>, arg: &clap::Arg) -> io::Result<()
         Some([]) => unreachable!(),
         None => arg.get_id().to_string().to_ascii_uppercase(),
     };
+    let (formatted_value_name, optional) = match arg.is_required_set() {
+        true => (format!("<{value_name}>"), ""),
+        false => (format!("[{value_name}]"), "(optional)"),
+    };
 
     match (arg.get_short(), arg.get_long()) {
         (Some(short), Some(long)) => {
             if arg.get_action().takes_values() {
-                write!(buffer, "`-{short}`, `--{long} <{value_name}>`")?
+                write!(
+                    buffer,
+                    "`-{short}`, `--{long} {formatted_value_name}` {optional}"
+                )?
             } else {
                 write!(buffer, "`-{short}`, `--{long}`")?
             }
         }
         (Some(short), None) => {
             if arg.get_action().takes_values() {
-                write!(buffer, "`-{short} <{value_name}>`")?
+                write!(buffer, "`-{short} {formatted_value_name}` {optional}")?
             } else {
                 write!(buffer, "`-{short}`")?
             }
         }
         (None, Some(long)) => {
             if arg.get_action().takes_values() {
-                write!(buffer, "`--{} <{value_name}>`", long)?
+                write!(buffer, "`--{} {formatted_value_name}`", long)?
             } else {
                 write!(buffer, "`--{}`", long)?
             }
         }
         (None, None) => {
-            write!(buffer, "`<{value_name}>`",)?;
+            write!(buffer, "`{formatted_value_name}` {optional}")?;
         }
     }
```

Comment:
I was testing this and I noticed a pattern. The `(optional)` text should be in all `write!()` calls, regardless of whether we are showing the `formatted_value_name` or not. It would seem that the `(optional)` text is actually a constant in every branch, so we could actually have the following:

```rust
match (arg.get_short(), arg.get_long()) {
    // You just include the `<{value_name}>` by 
    // `{formatted_value_name}` replacement where needed
    ...
}
// In this case though, the `optional` variable should include the initial whitespace
write!(buffer, "{optional}")?;
```

Let me know if that makes sense to you!

Reply:
Makes sense to me! I've made those changes and it's gone from 
<img width="817" alt="image" src="https://github.com/build-trust/ockam/assets/22850972/5b734bc8-33d7-444a-998d-a7d50ab9f026">
to
<img width="814" alt="image" src="https://github.com/build-trust/ockam/assets/22850972/c6d3893a-3298-48b5-a096-b951745f8415">